Lib/ast.py

```diff
@@ -117,6 +117,14 @@ def dump(node, annotate_fields=True, include_attributes=False, *, indent=None):
     integer or string, then the tree will be pretty-printed with that indent
     level. None (the default) selects the single line representation.
     """
+    def _qualifier_to_default(qualifier):
+        if qualifier == 1:
+            return []
+        elif qualifier == 2:
+            return None
```

Comment:
Would be nice to use an enum here as well, as in the c code instead of 1 and 2

Reply:
Maybe a c function that takes the qualifier and returns a default value for it? The the interpretation of the qualifier meaning is completely in c.

Reply:
I initially thought about creating an `IntEnum` but then it bothered me to since this was going to be a single use thing in the whole codebase. I didn't think much about the second suggestion, but from what I see it makes sense. Thanks!

Reply:
After a bit thinking, IMHO adding a converter to such a case in C and linking it in the `_ast` will have some cost, since the current system designed for only node classes. It might make sense to introduce an `IntEnum` though, but we still need to be sure whether the `_field_qualifiers` will be kept as private property or be public just like `_fields`.

Reply:
Do you actually use the value returned from _qualifier_to_default other than to check whether is None?

Reply:
Short answer: I will.

This PR actually splitted up from another PR (which can be seen from the bugtracker). The initial version, keeps the compatability with the ast.dump by only ignoring None's. The next patch I'll sent will ignore both Nones and empty lists (like ClassDef(bases=[], decorators[], body=[Pass()]) would be ClassDef(body=[])) with a special paramater on ast.dump (omit_defaults)